04/jatinsinghal1921/usertweets.py

```diff
@@ -0,0 +1,85 @@
+from collections import namedtuple
+import csv
+import os
+
+import tweepy
+
+from config import CONSUMER_KEY, CONSUMER_SECRET
+from config import ACCESS_TOKEN, ACCESS_SECRET
+
+DEST_DIR = 'data'
+EXT = 'csv'
+NUM_TWEETS = 100
+
+Tweet = namedtuple('Tweet', 'id_str created_at text')
+
+
+class UserTweets(object):
+
+    def __init__(self, handle, max_id=None):
+        """Get handle and optional max_id.
+        Use tweepy.OAuthHandler, set_access_token and tweepy.API
+        to create api interface.
+        Use _get_tweets() helper to get a list of tweets.
+        Save the tweets as data/<handle>.csv"""
+        # ...
+
+        auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
+        auth.set_access_token(ACCESS_TOKEN, ACCESS_SECRET)
+        self.api = tweepy.API(auth)
+        
+        self.handle = handle
+        self.max_id = max_id
+
+        self._tweets = list(self._get_tweets())
+        self.output_file = self._save_tweets()
+
+
+    def _get_tweets(self):
+        """Hint: use the user_timeline() method on the api you defined in init.
+        See tweepy API reference: http://docs.tweepy.org/en/v3.5.0/api.html
+        Use a list comprehension / generator to filter out fields
+        id_str created_at text (optionally use namedtuple)"""
+        tweets = (self.api).user_timeline(screen_name = self.handle, count = NUM_TWEETS, since_id = self.max_id)
+        return tweets
+
+
+    def _save_tweets(self):
+        """Use the csv module (csv.writer) to write out the tweets.
+        If you use a namedtuple get the column names with Tweet._fields.
+        Otherwise define them as: id_str created_at text
+        You can use writerow for the header, writerows for the rows"""
+        # file_name = self.handle + ".csv"
+        if not os.path.exists(DEST_DIR):
+            os.makedirs(DEST_DIR)
+        
+        file_name =  "./" + DEST_DIR + "/" + self.handle + ".csv"
+        with open(file_name, mode='w') as file_obj:
+            writer = csv.writer(file_obj, delimiter=',')
+            writer.writerow(['id_str', 'created_at', 'text'])
+
+            for tweet in self._tweets:
+                writer.writerow([tweet.id_str, tweet.created_at, tweet.text])
+        
+        return file_name
+
+
+    def __len__(self):
+        """See http://pybit.es/python-data-model.html"""
+        return len(self._tweets)
+
+
+    def __getitem__(self, pos):
+        """See http://pybit.es/python-data-model.html"""
+        return (self._tweets)[pos]
```

Comment:
Consider adding some type checks? As per https://docs.python.org/3/reference/datamodel.html#object.__getitem__ you can throw a TypeError here.